From e8b9ccfabd84ef07dec9d9af6241cd5352597640 Mon Sep 17 00:00:00 2001
From: Fang Jia <fang.jia@windriver.com>
Date: Sat, 14 May 2016 13:45:52 +0800
Subject: [PATCH] lttng commands stuck after repetitive restarts with over the
 network traces

issue: LINCCM-627

Add KEEPALIVE on lttng-relayd daemon socket in case the client
socket is abort.

Signed-off-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 src/bin/lttng-relayd/main.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/src/bin/lttng-relayd/main.c b/src/bin/lttng-relayd/main.c
index 0c7ef8d..6a96bf0 100644
--- a/src/bin/lttng-relayd/main.c
+++ b/src/bin/lttng-relayd/main.c
@@ -900,6 +900,13 @@ restart:
 					lttcomm_destroy_sock(newsock);
 					goto error;
 				}
+				ret = setsockopt(newsock->fd, SOL_SOCKET, SO_KEEPALIVE, &val,
+						 sizeof(val));
+				if (ret < 0) {
+					PERROR("setsockopt inet keepalive");
+					lttcomm_destroy_sock(newsock);
+					goto error;
+				}
 				new_conn = connection_create(newsock, type);
 				if (!new_conn) {
 					lttcomm_destroy_sock(newsock);
-- 
1.9.1

